src/modules/CRUD/components/Details/Create/Create.js

```diff
@@ -0,0 +1,194 @@
+ * @returns {Object} JSON schema properties cleaned
+ */
+const cleanProperties = props => {
+  const propertiesWithoutGroup = Object.values(props).reduce((list, value) => {
+    const { type, properties, required = [] } = value;
+
+    if (type === 'object' && required.length) {
+      const requiredProps = Object.entries(properties).reduce((acc, [subKey, subValue]) => (
+        required.includes(subKey)
+          ? ({ [subKey]: subValue })
+          : acc
+      ), {});
+
+      return {
+        ...list,
+        properties: { ...list.properties, ...requiredProps },
+        required: [...list.required, ...required],
+      };
+    }
+
+    return list;
+  }, { properties: {}, required: [] });
+
+  return propertiesWithoutGroup;
+};
+
+/**
+ * Build a json schema object
+ *
+ * @param {Object} {
+ * schemaProperties {Object},
+ * uiSchemaProperties {Object},
+ * geomType {Number}
+ *  }
+ * @returns an object compatible with jsonSchema specification
+ */
+const buildSchema = ({ schemaProperties, uiSchemaProperties, geomType }) => {
+  const {
+    schema: geomSchema,
+    ui_schema: geomUiSchema,
+    value,
+  } = getJSONSchemaFromGeom({ identifier: null, geom: null, title: 'Geométrie', geom_type: geomType });
+
+  const { required, properties } = cleanProperties(schemaProperties);
+
+  const { 'ui:order': uiOrder = [] } = uiSchemaProperties;
+
+  return {
+    schema: {
+      properties: { geom: { ...geomSchema, default: value }, ...properties },
+      required: ['geom', ...required],
+      type: 'object',
+    },
+    uiSchema: {
+      geom: geomUiSchema,
+      ...uiSchemaProperties,
+      'ui:order': ['geom', ...uiOrder],
+    },
+  };
+};
+
+const Create = props => {
+  const {
+    history: { push },
+    view: {
+      featureEndpoint,
+      formSchema: { properties: schemaProperties } = {},
+      uiSchema: uiSchemaProperties,
+      layer: { geom_type: geomType, name },
+    },
+    saveFeature,
+    t,
+  } = props;
+
+  const isMounted = useRef(true);
+
+  const [loading, setLoading] = useState(false);
+  const [schema, setSchema] = useState({});
+  const [uiSchema, setUiSchema] = useState({});
+
+  useEffect(() => {
+    isMounted.current = true;
+    return () => { isMounted.current = false; };
+  }, []);
+
+  useMemo(() => {
+    const builder = buildSchema({ schemaProperties, uiSchemaProperties, geomType });
+    setSchema(builder.schema);
+    setUiSchema(builder.uiSchema);
+  }, [geomType, schemaProperties, uiSchemaProperties]);
+
+
+  const handleSubmit = useCallback(async ({ formData: { geom, ...properties } }) => {
+    setLoading(true);
+
+    setSchema(prevSchema => {
+      const defaultValues = Object.entries(properties);
+      if (defaultValues.length) {
+        const nextSchema = { ...prevSchema };
+        defaultValues.forEach(([key, value]) => {
+          nextSchema.properties[key].default = value;
+        });
+        return nextSchema;
+      }
+      return prevSchema;
```

Comment:
This could be a dedicated (pure) function with `properties` and `prevSchema` as arguments. The name of this function would then *explain* what transformation is done.

Reply:
Right, done.